src/components/CourseView/index.tsx

```diff
@@ -40,7 +40,7 @@ const CourseView = () => {
   const [courses, coursesDispath] = useContext(CoursesContext);
   const [filters, filtersDispatch] = useContext(FilterContext);
 
-  const filteredCourses = useMemo(
+  const filteredCourses = useCallback(
```

Comment:
This function should be `getFilteredCourses` and your `getFilteredCourses` implementation can be moved to `./helpers.ts` and then imported as:
```tsx
import { getFilteredCourses as getFilteredCoursesHelper } from './helpers';
```